grpc-sys/build.rs

```diff
@@ -254,7 +250,18 @@ fn setup_openssl(config: &mut Config) {
 }
 
 #[cfg(not(feature = "openssl-vendored"))]
-fn setup_openssl(_config: &mut Config) {}
+fn setup_openssl(config: &mut Config) {
+    // check if openssl provided from system support ALPN
+    if Build::new()
+        .file("grpc/test/build/openssl-alpn.c")
+        .cargo_metadata(false)
+        .cpp(true)
+        .try_compile("check_alpn")
```

Comment:
How about convert to `Tool` and then compile? Seems call `try_compile` directly will link the output to the crate.

Reply:
`cargo_metadata(false)` will prevent that thing

Reply:
Got it.